sotodlib/site_pipeline/make_filterbin_map.py

```diff
@@ -0,0 +1,1113 @@
+    
+        # Detrend, subtract hwpss
+        detrend_tod(obs, method='median', signal_name='signal')
+        if remove_hwpss:
+            subtract_hwpss(obs)
+            signal_name = 'hwpss_remove'
+        else:
+            signal_name = 'signal'
+        
+        # Trending cuts
+        get_trending_cuts(obs)
+
+        # Jump detection
+        get_jumps(obs, signal_name=signal_name)
+    
+        # Glitches detection
+        get_glitches(obs, signal_name=signal_name)
+
+        # Detrend
+        if remove_hwpss:
+            detrend_tod(obs, method='median', signal_name=signal_name)
+    
+        # P2P cuts
+        ptp_cuts(obs, signal_name)
+    return True
+
+
+def cal_pW(obs, signal_name):
+    obs.signal = np.multiply(obs.signal.T, obs.det_cal.phase_to_pW).T
+    if signal_name == 'hwpss_remove':
+        obs.hwpss_remove = np.multiply(obs.hwpss_remove.T, obs.det_cal.phase_to_pW).T
+
+        
+def cal_rel(obs, signal_name):
+    # LPF + PCA
+    filt = filters.low_pass_sine2(1, width=0.1)
+    sigfilt = filters.fourier_filter(obs, filt, signal_name=signal_name)
+    obs.wrap(f'lpf_{signal_name}', sigfilt, [(0,'dets'),(1,'samps')])
+    obs.restrict('samps',(10*400, -10*400))
+
+    pca_out = pca.get_pca(obs, signal=obs[f'lpf_{signal_name}'])
+    pca_signal = pca.get_pca_model(obs, pca_out, signal=obs[f'lpf_{signal_name}'])
+
+    med = np.median(pca_signal.weights[:,0])
+    
+    # Relative calib
+    obs[f'{signal_name}'] = np.divide(obs.signal.T, pca_signal.weights[:,0]/med).T
+
+    
+def calibrate_data(obs, remove_hwpss=True):
+    if remove_hwpss:
+        signal_name = 'hwpss_remove'
+    else:
+        signal_name = 'signal'
+
+    # Get data in pW 
+    cal_pW(obs, signal_name=signal_name)
+    
+    # Get relative calibrated data
+    cal_rel(obs, signal_name=signal_name)
+
+    # Get absolute calibrated data
+    obs[f'{signal_name}'] = np.multiply(obs[f'{signal_name}'].T,
+                                        obs.abscal.abscal_factor).T
+    
+    return obs
+
+
+def readout_filter(obs, remove_hwpss=True):
+    if remove_hwpss:
+        signal_name = 'hwpss_remove'
+    else:
+        signal_name = 'signal'
+        
+    iir_par = obs.iir_params[f'ufm_{obs.det_info.wafer.array[0]}']
+    if iir_par['a'] is None or iir_par['b'] is None:
+        return False
+    filt = filters.iir_filter(iir_params=iir_par, invert=True)
+
+    obs[signal_name] = filters.fourier_filter(obs, filt, signal_name=signal_name)
+    
+    return True
+
+
+def deconvolve_detector_tconst(obs, remove_hwpss=True):
+    if remove_hwpss:
+        signal_name = 'hwpss_remove'
+    else:
+        signal_name = 'signal'
+
+    filt = filters.timeconst_filter(timeconst = obs.det_cal.tau_eff,
+                                        invert=True)
+    
+    obs[signal_name] = filters.fourier_filter(obs, filt, signal_name=signal_name)
+        
+        
+def demodulate_hwp(obs):
+    apodize.apodize_cosine(obs)
+    hwp.demod_tod(obs)
```

Comment:
This function by default will demodulate `obs.signal` but all the processing has been done over `obs[f'{signal_name}']`. They might not be the same.

Reply:
Done